pkg/certificate/manager.go

```diff
@@ -44,24 +46,21 @@ func (m *Manager) start(checkInterval time.Duration) {
 	// iterate over the list of certificates
 	// when a cert needs to be rotated - call RotateCertificate()
 	ticker := time.NewTicker(checkInterval)
-	done := make(chan bool)
-	m.stopRotation = done
 	go func() {
 		for {
 			select {
-			case <-done:
+			case <-m.certRotation:
 				ticker.Stop()
 				return
-			default:
+			case <-ticker.C:
 				m.checkAndRotate()
-				<-ticker.C
 			}
 		}
 	}()
 }
 
 func (m *Manager) stop() {
-	m.stopRotation <- true
+	<-m.certRotation
 }
```

Comment:
When this needs to stop, callers can `close()` the channel passed in when this was initialized, so we don't need a separate function for this.